### PR TITLE
feat: add the ability to auth in using certificate and private key

### PR DESCRIPTION
**Pull request by DiRaiks**
**What's the problem this PR addresses?**
Current .yarnrc.yml configuration does not support authorization in registry via certificate and private key.

...

**How did you fix it?**
Added the ability to specify values ​​for the certificate and private key. 
Added setting of certificate and key in configuration "got" package

...

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

**Comment by arcanis** [comment]
Using just `cert` and `key` as name doesn't match the naming convention from other settings - using raw values also seems prone to make the yarnrc file quite verbose (and doesn't match the syntax from their sibling, `caFilePath`). What do you think of turning those settings into this:?

```
httpsCertFilePath
httpsKeyFilePath
```

**Comment by DiRaiks** [comment]
@arcanis Thank you for your review!
Okay, I renamed the fields, changed their type to use the file path, and added caching.

File: .yarn/versions/ed770662.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "@yarnpkg/plugin-patch": minor
+  "@yarnpkg/plugin-pnp": minor
+  "@yarnpkg/plugin-pnpm": minor
+  "@yarnpkg/plugin-stage": minor
+  "@yarnpkg/plugin-typescript": minor
+  "@yarnpkg/plugin-version": minor
+  "@yarnpkg/plugin-workspace-tools": minor
+  "@yarnpkg/pnp": minor
+  "@yarnpkg/pnpify": minor
+  "@yarnpkg/sdks": minor
```
Comment (paul-soporan) [inline .yarn/versions/ed770662.yml:33]: Only the core and the CLI need to be released, the others can be declined.

**Comment by paul-soporan** [comment]
> What do you think of turning those settings into this:?

Shouldn't `caFilePath` get renamed to `httpsCaFilePath` for consistency in that case?

**Comment by DiRaiks** [comment]
@paul-soporan Thank you for your review!
I renamed the field - `caFilePath` to `httpsCaFilePath` and change realease file

**Comment by paul-soporan** [comment]
The comment about renaming `caFilePath` was meant more as a way to start a discussion about what we should do to maintain consistency rather than something that needs to be worked on right away.

In this case, if we decide to rename it, we need to deprecate `caFilePath` and remove it from the docs before completely removing it in the next major.

While still on the topic of https option naming, is there any value to shortening the names, especially since the got options use the full names (`certificate` & `certificateAuthority`)?

**Comment by arcanis** [comment]
> While still on the topic of https option naming, is there any value to shortening the names, especially since the got options use the full names (`certificate` & `certificateAuthority`)?

Not sure about this one (no particular reason except that the long names "feel" very verbose), but yeah, `httpsCaFilePath` would be a good candidate for 4.0.

**Comment by arcanis** [comment]
Good work, thanks @DiRaiks !

**Comment by ctr1key** [comment]
I tested these changed in an environment with a proxy and self signed certificates and one more change seems to be necessary.
The certificate authority file must go in the called to tunnel.httpsOverHttp at line https://github.com/DiRaiks/berry/blob/55b6db7380dcbfca0e5f1249b9bcfe0675ce0f55/packages/yarnpkg-core/sources/httpUtils.ts#L237

See https://github.com/koichik/node-tunnel#https-over-https-tunneling for where the CA file goes on tunnel.httpsOverHttp. To avoid confusion, in my environment, 'ca' has to go on the same object as 'proxy', not the one inside 'proxy'. There are probably a few other settings from the networkConfig that would apply to tunnel.httpsOverHttp as well.

Should I create a new issue for this?

**Pull request merged**